components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java

```diff
@@ -331,6 +331,27 @@ public API getAPI(APIIdentifier identifier) throws APIManagementException {
         }
     }
 
+    /**
+     * Get API by registry artifact id
+     *
+     * @param uuid  Registry artifact id
+     * @return API of the provided artifact id
+     * @throws APIManagementException
+     */
+    public API getAPIbyUUID(String uuid) throws APIManagementException {
+        try {
+            //registry of the current logged in user has been directly used
+            GenericArtifactManager artifactManager = APIUtil.getArtifactManager(registry, APIConstants.API_KEY);
+
+            GenericArtifact apiArtifact = artifactManager.getGenericArtifact(uuid);
+            return APIUtil.getAPIForPublishing(apiArtifact, registry);
+
+        } catch (RegistryException e) {
+            handleException("Failed to get API", e);
```

Comment:
handlerException method thrown an Exception right? If so the return statement below this line of code is never reachable. We can remove it.